### ISSUE TITLE
Date parser: support minimum and maximum dates check.

### ISSUE THREAD

**Issue by scammi** [StefanTerdell/zod-to-json-schema#71]
The current date parser resolves very basic information from the ZOD schema, the type and format.

Our current usage of this library would greatly be improved by being able to support `min` and `max` checks. For ZOD documentation reference see [here](https://zod.dev/?id=dates).

**Comment by StefanTerdell** [comment]
Hmm, really not too keen about this one. You're using [reserved keywords](https://json-schema.org/understanding-json-schema/reference/numeric.html#id7) and there's no equivalent in the JSON schema spec.

**Comment by scammi** [comment]
Hey, where can I check for the JSON schema spec. And specifically, where is the reserved keyword being used ?

Thank you !

**Comment by StefanTerdell** [comment]
@scammi I linked the spec in my previous comments. `maximum` and `minimum` are reserved for `number` and `integer` type schemas.

**Comment by scammi** [comment]
I see what you mean.

We can conform to the specification if we follow this format.

```json
{
  "type": "integer",
  "format": "unix-time",
  "minimum": 1577836800,   // Minimum value (e.g., January 1, 2020)
  "maximum": 1672444799    // Maximum value (e.g., December 31, 2022)
}

```

**Comment by StefanTerdell** [comment]
Feature released in 3.21.2
